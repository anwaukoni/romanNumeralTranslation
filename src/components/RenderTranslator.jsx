import React from 'react';
import PropTypes from 'prop-types';
import { Label, Input, FormGroup } from 'reactstrap';

export default class RenderTranslator extends React.Component {
  constructor(props, context){
    super(props, context)

    this.state = {
      query: ''
    }

    this._translateQuery = this._translateQuery.bind(this);
  }

  _updateQuery(event) {
    this.setState({ query: event.target.value });
  }

  _translateQuery() {
    this.props.onTranslate(this.state.query);
    this.setState({ query: '' });
  }

  render() {
    const { name, type, placeholder, buttonText, label } = this.props;

    return (
      <FormGroup style={{ backgroundColor: '#e2f0c6', borderRadius: '4px', padding: '12px'}}>
        <Label for={name}>{ label }</Label>
        <div>
          <div style={{ display: 'inline-block', paddingRight: '16px', paddingBottom: '12px', verticalAlign: 'top', width: '60%' }}>
            <Input
              name={name}
              type={type}
              placeholder={placeholder}
              value={this.state.query}
              onChange={ event => this._updateQuery(event) }
              aria-label={name}
              aria-describedby="basic-addon2"
            />
          </div>
            <button 
              style={{
                color: '#fff',
                backgroundColor: '#337ab7',
                borderColor: '#2e6da4',
                display: 'inline-block',
                marginBottom: 0,
                fontWeight: 400,
                textAlign: 'center',
                verticalAlign: 'middle',
                padding: '6px 12px',
                fontSize: '14px',
                lineHeight: '1.42857143',
                borderRadius: '4px'
              }}  
              onClick={this._translateQuery}
            >
              { buttonText }
            </button>
        
        </div>
      </FormGroup>
    );
  }
}

RenderTranslator.propTypes = {
  onTranslate: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired
};