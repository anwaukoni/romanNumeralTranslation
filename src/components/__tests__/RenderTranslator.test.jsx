import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import RenderTranslator from '../RenderTranslator';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Given the RenderTranslator component', () => {
  describe('when rendering a translator', () => {
    it('should render in the correct format', () => {
      const wrapper = shallow(
        <RenderTranslator
          name='translator'
          type='text'
          placeholder='translator'
          buttonText='Translate To Number'
          onTranslate={() => {}}
          label='Roman Numerals To Number'
        />
      );
        
      expect(wrapper).toMatchSnapshot();
    });
  });
});
