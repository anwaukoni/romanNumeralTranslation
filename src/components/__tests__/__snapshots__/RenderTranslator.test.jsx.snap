// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the RenderTranslator component when rendering a translator should render in the correct format 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RenderTranslator
    buttonText="Translate To Number"
    label="Roman Numerals To Number"
    name="translator"
    onTranslate={[Function]}
    placeholder="translator"
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          for="translator"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Roman Numerals To Number
        </Label>,
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "paddingBottom": "12px",
                "paddingRight": "16px",
                "verticalAlign": "top",
                "width": "60%",
              }
            }
          >
            <Input
              aria-describedby="basic-addon2"
              aria-label="translator"
              name="translator"
              onChange={[Function]}
              placeholder="translator"
              type="text"
              value=""
            />
          </div>
          <button
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#337ab7",
                "borderColor": "#2e6da4",
                "borderRadius": "4px",
                "color": "#fff",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 400,
                "lineHeight": "1.42857143",
                "marginBottom": 0,
                "padding": "6px 12px",
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Translate To Number
          </button>
        </div>,
      ],
      "style": Object {
        "backgroundColor": "#e2f0c6",
        "borderRadius": "4px",
        "padding": "12px",
      },
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Roman Numerals To Number",
          "for": "translator",
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": "Roman Numerals To Number",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                Object {
                  "display": "inline-block",
                  "paddingBottom": "12px",
                  "paddingRight": "16px",
                  "verticalAlign": "top",
                  "width": "60%",
                }
              }
            >
              <Input
                aria-describedby="basic-addon2"
                aria-label="translator"
                name="translator"
                onChange={[Function]}
                placeholder="translator"
                type="text"
                value=""
              />
            </div>,
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#337ab7",
                  "borderColor": "#2e6da4",
                  "borderRadius": "4px",
                  "color": "#fff",
                  "display": "inline-block",
                  "fontSize": "14px",
                  "fontWeight": 400,
                  "lineHeight": "1.42857143",
                  "marginBottom": 0,
                  "padding": "6px 12px",
                  "textAlign": "center",
                  "verticalAlign": "middle",
                }
              }
            >
              Translate To Number
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Input
                aria-describedby="basic-addon2"
                aria-label="translator"
                name="translator"
                onChange={[Function]}
                placeholder="translator"
                type="text"
                value=""
              />,
              "style": Object {
                "display": "inline-block",
                "paddingBottom": "12px",
                "paddingRight": "16px",
                "verticalAlign": "top",
                "width": "60%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-describedby": "basic-addon2",
                "aria-label": "translator",
                "name": "translator",
                "onChange": [Function],
                "placeholder": "translator",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Translate To Number",
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "#337ab7",
                "borderColor": "#2e6da4",
                "borderRadius": "4px",
                "color": "#fff",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 400,
                "lineHeight": "1.42857143",
                "marginBottom": 0,
                "padding": "6px 12px",
                "textAlign": "center",
                "verticalAlign": "middle",
              },
            },
            "ref": null,
            "rendered": "Translate To Number",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            for="translator"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Roman Numerals To Number
          </Label>,
          <div>
            <div
              style={
                Object {
                  "display": "inline-block",
                  "paddingBottom": "12px",
                  "paddingRight": "16px",
                  "verticalAlign": "top",
                  "width": "60%",
                }
              }
            >
              <Input
                aria-describedby="basic-addon2"
                aria-label="translator"
                name="translator"
                onChange={[Function]}
                placeholder="translator"
                type="text"
                value=""
              />
            </div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#337ab7",
                  "borderColor": "#2e6da4",
                  "borderRadius": "4px",
                  "color": "#fff",
                  "display": "inline-block",
                  "fontSize": "14px",
                  "fontWeight": 400,
                  "lineHeight": "1.42857143",
                  "marginBottom": 0,
                  "padding": "6px 12px",
                  "textAlign": "center",
                  "verticalAlign": "middle",
                }
              }
            >
              Translate To Number
            </button>
          </div>,
        ],
        "style": Object {
          "backgroundColor": "#e2f0c6",
          "borderRadius": "4px",
          "padding": "12px",
        },
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Roman Numerals To Number",
            "for": "translator",
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": "Roman Numerals To Number",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "paddingBottom": "12px",
                    "paddingRight": "16px",
                    "verticalAlign": "top",
                    "width": "60%",
                  }
                }
              >
                <Input
                  aria-describedby="basic-addon2"
                  aria-label="translator"
                  name="translator"
                  onChange={[Function]}
                  placeholder="translator"
                  type="text"
                  value=""
                />
              </div>,
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#337ab7",
                    "borderColor": "#2e6da4",
                    "borderRadius": "4px",
                    "color": "#fff",
                    "display": "inline-block",
                    "fontSize": "14px",
                    "fontWeight": 400,
                    "lineHeight": "1.42857143",
                    "marginBottom": 0,
                    "padding": "6px 12px",
                    "textAlign": "center",
                    "verticalAlign": "middle",
                  }
                }
              >
                Translate To Number
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Input
                  aria-describedby="basic-addon2"
                  aria-label="translator"
                  name="translator"
                  onChange={[Function]}
                  placeholder="translator"
                  type="text"
                  value=""
                />,
                "style": Object {
                  "display": "inline-block",
                  "paddingBottom": "12px",
                  "paddingRight": "16px",
                  "verticalAlign": "top",
                  "width": "60%",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-describedby": "basic-addon2",
                  "aria-label": "translator",
                  "name": "translator",
                  "onChange": [Function],
                  "placeholder": "translator",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Translate To Number",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "#337ab7",
                  "borderColor": "#2e6da4",
                  "borderRadius": "4px",
                  "color": "#fff",
                  "display": "inline-block",
                  "fontSize": "14px",
                  "fontWeight": 400,
                  "lineHeight": "1.42857143",
                  "marginBottom": 0,
                  "padding": "6px 12px",
                  "textAlign": "center",
                  "verticalAlign": "middle",
                },
              },
              "ref": null,
              "rendered": "Translate To Number",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
